sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -183,6 +184,43 @@ case class OptionBooleanAggregator(colName: String)
   def OptionalBoolEncoder: Encoder[Option[Boolean]] = ExpressionEncoder()
 }
 
+case class OptionBooleanIntAggregator(colName: String)
+    extends Aggregator[Row, Option[(Boolean, Int)], Option[(Boolean, Int)]] {
```

Comment:
what's the expected schema after we apply an aggregator with `Option[Product]` as buffer/output?

Reply:
assuming non top level, `Option[Product]` is same as `Product`?

Reply:
Yes. For non top level, `[Option[Product]` is same as `Product`. The difference is additional `WrapOption` and `UnwrapOption` around expressions.